tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -467,6 +467,98 @@ static PyObject * THPVariable_get_device(PyObject* self_, PyObject* args, PyObje
 
 static PyObject * THPVariable_numel(PyObject* self_, PyObject* args, PyObject* kwargs);
 
+// linspace
+static PyObject * THPVariable_linspace(PyObject* self_, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    "linspace(Scalar start, Scalar end, int64_t? steps=None, *, Tensor out=None, ScalarType dtype=None, Layout layout=torch.strided, Device device=None, bool pin_memory=False, bool requires_grad=False)",
+  }, /*traceable=*/true);
+
+  ParsedArgs<9> parsed_args;
+  auto _r = parser.parse(nullptr, args, kwargs, parsed_args);
+  if(_r.has_torch_function()) {
+    return handle_torch_function(_r, nullptr, args, kwargs, THPVariableFunctionsModule, "torch");
+  }
+  if (_r.isNone(3)) {
+    // aten::linspace(Scalar start, Scalar end, int? steps=None, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor
+
+    // NOTE: r.scalartype(X) gives the default dtype if r.isNone(X)
```

Comment:
add another line explicitly stating that it is not desirable for the case when one or both of start and end values are complex numbers but the dtype is None